sql/src/test/java/io/crate/operation/scalar/conditional/ConditionalFunctionTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.conditional;
+
+import io.crate.analyze.symbol.Literal;
+import io.crate.operation.Input;
+import io.crate.operation.scalar.AbstractScalarFunctionsTest;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ConditionalFunctionTest extends AbstractScalarFunctionsTest {
```

Comment:
pls checkout the other child implementation of AbstractScalarFunctionTest, all can be done much easier with less code ;)